### ISSUE TITLE
Engine crashes if ZSS compiler encounters a helper sctrl with { on separate line

### ISSUE THREAD

**Issue by Jmorphman** [ikemen-engine/Ikemen-GO#1827]
### Describe the bug

Pretty much exactly what is described in the title. Assume there's some .zss file that IKEMEN is trying to compile, if it encounters the following:
```
helper {
    stateno: 12345;
    id: 12345;
}
```
then it will correctly interpret the code, no issues whatsoever. However, if we make a tiny modification, moving the helper sctrl's opening bracket to a newline:
```
helper
{
    stateno: 12345;
    id: 12345;
}
```
then the engine will crash upon encountering this; further, it will not display any error messages, or output anything to Ikemen.log

### To Reproduce

Add a Helper sctrl to any .ZSS file, making sure the opening bracket is on a subsequent line after `Helper` itself, and then have IKEMEN try to load and compile that file.

### Expected behavior

For all other state controllers, and indeed all other instances of opening and closing brackets, there is no requirement that the opening bracket must be placed on the same line as whatever state controller or etc. requires it.

While this is ultimately a pretty minor concern, the lack of any user feedback as to the nature of the crash probably warrants action.

### Engine Version (or source code date)

2024.06.03

### Operating system

Windows

**Comment by potsmugen** [comment]
Interesting. This is for sure because of the helper redirection. The code is expecting "," or "(" or "{" but not a line break.

**Comment by Jmorphman** [comment]
> Interesting. This is for sure because of the helper redirection. The code is expecting "," or "(" or "{" but not a line break.

Indeed; funnily enough, I happened upon this while doing some tests for syntax highlighting stuff, where newlines pose similar difficulties. Although, this is is a bit of an interesting case:
```
if c.token == "helper" && ((*line)[0] == ',' || (*line)[0] == '(') {
	ok = false
}
```
The issue here arises because the compiler is trying to check the next character in sequence on the line that it has found a string of text matching `helper`. If the `{` is placed on the next line in the ZSS file, then `line` at this point will be an empty string (even if there were whitespace characters following `helper`, because those would be trimmed and removed). A fatal error occurs when the compiler tries to access the first character in that empty string, causing the crash.

This specific case is easily fixed just by adding a check for the string length to be greater than 0, but those `(*line)[0]` still seem troublesome to me: suppose an author wrote `helper (456), blah blah` somewhere: the compiler would interpret this as a state controller, simply because the opening parentheses is not the next character in sequence after `helper`.

Then again, my extremely brief testing and examination of the code seems to indicate that the only time this situation were to occur is if this hypothetical author who put a space between the `helper` and parentheses were to have placed that statement as the first thing on a new line in their code—and I'm really not sure what kind of valid ZSS statement that could be the start of, so perhaps this check isn't needed at all? Though again, I'm not really familiar with the compiler code at all.

EDIT: turned out to be a fairly simple fix that preserves current behavior: #1831
